velox/functions/sparksql/tests/AtLeastNNonNullsTest.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/specialforms/AtLeastNNonNulls.h"
+#include "velox/common/base/tests/FloatConstants.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+using namespace facebook::velox::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class AtLeastNNonNullsTest : public SparkFunctionBaseTest {
+ public:
+  AtLeastNNonNullsTest() {
+    // Allow for parsing literal integers as INTEGER, not BIGINT.
+    options_.parseIntegerAsBigint = false;
+  }
+
+ protected:
+  template <typename T>
+  VectorPtr atLeastNNonNulls(
+      std::optional<T> n,
+      const std::vector<VectorPtr>& input,
+      bool useNullForN = false) {
```

Comment:
Why do we have both optional n and this boolean?

Reply:
We don't need this if we hard code the expression. Thanks.